rosa_tutorials/rosa-mobb-azure-group-claims.adoc

```diff
@@ -0,0 +1,149 @@
+
+[NOTE]
+====
+In this example, we are providing all security groups a user is a member of via the group claim. In a real production environment, we highly recommend _scoping the groups provided by the group claim to _only those groups which are applicable to OpenShift_.
+====
+
+image:azure-portal_edit-group-claims-page.png[Azure Portal - Edit Groups Claim Page]
+
+[id="rosa-mobb-azure-group-claims-configure-cluster"]
+== Configure the OpenShift cluster to use Azure AD as the identity provider
+
+Finally, we need to configure OpenShift to use Azure AD as its identity provider. While Red Hat OpenShift Service on AWS (ROSA) offers the ability to configure identity providers via the OpenShift Cluster Manager (OCM), that functionality does not currently support group claims. Instead, we'll configure the cluster's OAuth provider to use Azure AD as its identity provider via the `rosa` CLI. To do so, run the following command, making sure to replace the variable specified:
+
+[source,terminal]
+----
+$ rosa create idp \
+    --cluster ${CLUSTER_NAME} \ <1>
+    --type openid \
+    --name ${IDP_NAME} \ <2> 
+    --client-id ${APP_ID} \ <3> 
+    --client-secret ${CLIENT_SECRET} \ <4> 
+    --issuer-url https://login.microsoftonline.com/${TENANT_ID}/v2.0 \ <5>
+    --email-claims email \
+    --name-claims name \
+    --username-claims preferred_username \
+    --groups-claims groups \
+    --extra-scopes email,profile
+----
+--
+<1> Provide the name of your ROSA cluster with the `export CLUSTER_NAME=example-cluster` command in your terminal.
+<2> Provide the name you used in the OAuth callback URL with the `export IDP_NAME=AAD` command in your terminal.
+<3> Provide the Application (client) ID with the `export APP_ID=yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy` command in your terminal.
+<4> Provide the client secret with the `export CLIENT_SECRET=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx` command in your terminal.
+<5> Provide the Directory (tenant) ID with the `export TENANT_ID=zzzzzzzz-zzzz-zzzz-zzzz-zzzzzzzzzzzz` command in your terminal.
+--
+
+[id="rosa-mobb-azure-group-claims-additional-permissions"]
+== Grant additional permissions to individual groups
+
+Once the cluster authentication Operator reconciles your changes (generally within a few minutes), you will be able to login to the cluster using Azure AD. In addition, the cluster OAuth provider will automatically create or update the membership of groups the user is a member of (using the group ID).
+
+Once you login, you will notice that you have very limited permissions. This is because, by default, OpenShift only grants you the ability to create new projects (namespaces) in the cluster. Other projects (namespaces) are restricted from view. The cluster OAth provider **does not** automatically create RoleBindings and ClusterRoleBindings for the groups that are created, you are responsible for creating those via your own processes.
```

Comment:
```suggestion
Once you log in, you will notice that you have very limited permissions. This is because, by default, OpenShift only grants you the ability to create new projects (namespaces) in the cluster. Other projects (namespaces) are restricted from view. The cluster OAth provider **does not** automatically create RoleBindings and ClusterRoleBindings for the groups that are created, you are responsible for creating those via your own processes.
```